npm/src/commands/validateDefinitions.js

```diff
@@ -3,151 +3,83 @@
-      "'v<MAJOR>.<MINOR>.<PATCH>'. (Use 'x' as a wildcard where necessary)"
-    );
-    for (let i = 0; i < 3 - parts.length; i++) {
-      parts.push('x');
-    }
-    ver ='v' + parts.join('.');
-  }
-
-  // Each part must either be a number, or must be 'x'
-  parts.forEach((part, idx) => {
-    if (part !== 'x' && (String(parseInt(part, 10)) !== part)) {
-      const partName = VERSION_PART_IDX_TO_NAME[idx];
-      errors.push(
-        "'" + prefix + givenVer + "': '" + part + "' is not a valid " +
-        partName + " version!"
-      );
-    }
-  });
-
-  // The 'major' portion of a version cannot be a wildcard
-  if (parts[0] === 'x') {
-    errors.push(
-      "'" + givenVer + "': The 'major' part of a version cannot be 'x'!"
-    );
-  }
-
-  return errors;
-}
-
 async function validateDefinitions(): Promise<Map<string, Array<string>>> {
-  const pkgNames = await getPkgNames();
-  const pkgNameVersions = await P.all(pkgNames.map(getPkgVersions));
-
   const results = new Map();
-  await P.all(pkgNameVersions.map(async (pkgVersions, idx) => {
-    const pkgName = pkgNames[idx];
-    await P.all(pkgVersions.map(async (pkgVer, idx) => {
-      const pkgNameVer = pkgName + "@" + pkgVer;
-      let errs = results.get(pkgNameVer) || [];
-
-      // Validate version string
-      validatePkgVersion(pkgVer).forEach(err => errs.push(err));
 
-      const pkgVerDir = path.join(DEFINITIONS_DIR, pkgName, pkgVer);
-      const pkgVerDirItems = await fs.readdir(pkgVerDir);
-      const pkgVerDirItemStats = await P.all(pkgVerDirItems.map(
-        itemName => fs.stat(path.join(pkgVerDir, itemName))
-      ));
-      pkgVerDirItems.forEach((itemName, idx) => {
-        const itemStats = pkgVerDirItemStats[idx];
-        if (isTestFileName(itemName, itemStats)) {
-          return;
+  const pkgVersions = await getPackages();
+  await P.all(pkgVersions.map(async (pkg) => {
+    const pkgVerStr = versionToString(pkg.version);
+    const pkgErrKey = pkg.name + '-' + pkgVerStr;
+    const errs = results.get(pkgErrKey) || [];
+
+    const dirItems = await fs.readdir(pkg.path);
+    await P.all(dirItems.map(async (itemName) => {
+      const itemPath = path.join(pkg.path, itemName);
+      const itemStats = await fs.stat(itemPath);
+      if (itemStats.isDirectory()) {
+        if (itemName.substr(0, 'flow-'.length) !== 'flow-'
+            || !VERSIONED_NAME_RE.test(itemName)) {
+          errs.push(
+            itemPath + ": All directories under a package-version dir must " +
+            "be named 'flow-VERSION'."
+          );
         }
-
-        const itemPath = path.join(pkgName, pkgVer, itemName);
-        if (itemStats.isFile()) {
+      } else if (itemStats.isFile()) {
+        if (!isTestFileName(itemName)) {
           errs.push(
-            "'" + itemPath + "': Only 'test-*' files are permitted in this " +
-            "directory!"
+            itemPath + ": All files under a package-version dir must be test " +
```

Comment:
Unfortunately they also seem to pick up all the whitespace after the newline -- so this doesn't come out very well formatted :(

```
asdf: Only 'test-*' files are permitted in this 
            directory!
```